source/MRViewer/ImGuiHelpers.cpp

```diff
@@ -353,125 +357,144 @@ bool BeginCustomStatePlugin( const char* label, bool* open, bool* collapsed, flo
         SetNextWindowPos( ImVec2( GetIO().DisplaySize.x - width, yPos ), ImGuiCond_FirstUseEver );
     }
 
-    SetNextWindowSize( ImVec2( width, height ) );
-    const auto constriants = ( height == 0.0f ) ? ImVec2{ width, -1.0f } : ImVec2{ width, height };
-    SetNextWindowSizeConstraints( constriants, constriants );
+    if ( changedSize )
+    {
+        if ( collapsed && *collapsed )
+        {
+            SetNextWindowSize( { changedSize->x, height }, ImGuiCond_Always );
+        }
+        else
+        {
+            SetNextWindowSize( *changedSize, ImGuiCond_Always );
+        }
+        
+    }
+    else
+        SetNextWindowSize( ImVec2( width, height ), ImGuiCond_Always );
 
-    //need no paddings if the window is collapsed
     if ( collapsed && *collapsed )
     {
-        ImGui::PushStyleVar( ImGuiStyleVar_WindowPadding, { 0, 0 } );
         ImGui::PushStyleVar( ImGuiStyleVar_WindowMinSize, { 0, 0 } );
     }
 
     if ( !Begin( label, open, flags | ImGuiWindowFlags_NoTitleBar | ImGuiWindowFlags_NoScrollbar | ImGuiWindowFlags_NoScrollWithMouse ) )
     {
         *open = false;
+        ImGui::PopStyleVar();
         if ( collapsed && *collapsed )
-            ImGui::PopStyleVar( 2 );
+            ImGui::PopStyleVar();
         return false;
     }
 
+    auto context = ImGui::GetCurrentContext();
+    window = context->CurrentWindow;
+
+    if ( changedSize && collapsed && !*collapsed )
+    {
+        changedSize->x = window->Rect().GetWidth();
+        changedSize->y = window->Rect().GetHeight();
+    }
+
     if ( collapsed && *collapsed )
-        ImGui::PopStyleVar( 2 );
+        ImGui::PopStyleVar();
 
     const auto bgColor = ImGui::ColorConvertFloat4ToU32(ImGui::GetStyleColorVec4( ImGuiCol_FrameBg ));
+  
 
-    auto context = ImGui::GetCurrentContext();
-    window = context->CurrentWindow;
-    const auto& style = ImGui::GetStyle();
-    ImGui::SetCursorPos( { 0, 0 } );
+    ImGui::PushStyleColor( ImGuiCol_Button, bgColor );
+    ImGui::PushStyleColor( ImGuiCol_Border, bgColor );
+    ImGui::PushStyleVar( ImGuiStyleVar_FrameBorderSize, 0.0f );
+    ImGui::PushStyleVar( ImGuiStyleVar_FramePadding, { MR::cRibbonItemInterval * scaling * 0.5f,  MR::cRibbonItemInterval * scaling * 0.5f } );
     
-    const ImVec2 pos = ImGui::GetCursorScreenPos();
-    ImFont* iconsFont = MR::RibbonFontManager::getFontByTypeStatic( MR::RibbonFontManager::FontType::Icons );;
-    ImFont* titleFont = MR::RibbonFontManager::getFontByTypeStatic( MR::RibbonFontManager::FontType::SemiBold );
-    if ( iconsFont )
+    const float buttonSize = titleBarHeight - MR::cRibbonItemInterval * scaling;
+    ImGui::SetCursorScreenPos( { window->Rect().Min.x + style.WindowPadding.x, window->Rect().Min.y  + closeButtonShift } );
+    
+    ImFont* iconsFont = nullptr;
+    ImFont* titleFont = nullptr;
+    if ( menu )
     {
+        iconsFont = menu->getFontManager().getFontByType( MR::RibbonFontManager::FontType::Icons );
```

Comment:
you can use static accessor to fonts now, and move menu cast under `if ( window )`